x-pack/legacy/plugins/epm/common/types.ts

```diff
@@ -30,34 +31,49 @@ export enum ElasticsearchAssetType {
   ilmPolicy = 'ilm-policy',
 }
 
-// Registry's response types
-// from /search
-// https://github.com/elastic/package-registry/blob/master/docs/api/search.json
-export type RegistryList = RegistryListItem[];
-export interface RegistryListItem {
-  description: string;
-  download: string;
-  icon: string;
+// from /package/{name}
+// type Package struct at https://github.com/elastic/package-registry/blob/master/util/package.go
+// https://github.com/elastic/package-registry/blob/master/docs/api/package.json
+export interface RegistryPackage {
   name: string;
-  version: string;
-  title?: string;
-}
-
-export interface ScreenshotItem {
-  src: string;
   title?: string;
+  version: string;
+  readme?: string;
+  description: string;
+  type: string;
+  categories: string[];
+  requirement: RequirementsByServiceName;
+  screenshots?: ScreenshotItem[];
+  icons?: string[];
+  assets?: string[];
+  internal?: boolean;
+  format_version: string;
+  datasets?: Dataset[];
```

Comment:
Was required (`datasets: Dataset[];`) now optional.

This was the source of errors when trying to install a package without datasets, like apache-1.0.1. Our code assumed it was always present and had a runtime error trying to iterate an undefined value.